linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -4402,21 +4408,28 @@ private[optimizer] object OptimizerCore {
    *  TODO It would be better for `Binding`s to represent the `this` reference
    *  using a separate mechanism.
    *
-   *  `this` is normally not a valid `LocalName`, because that is a JS keyword,
-   *  so we cast here.
+   *  We use an unsafely-constructed invalid `LocalName`, so that it never
+   *  clashes with normal local names. It is invalid because of the `;`.
+   */
+  private val LocalThisName =
+    LocalName.unsafeCreate("this;".map(_.toByte).toArray)
```

Comment:
I'm OK with this to unblock this PR, but I'd like to see it go away soon. Could you file a separate issue about this?

A concern I have is that a lot of code in `Encoding` makes the (reasonable) assumption that names are correct. In the current state, this wrong name will not break anything AFAICT but it might easily in the future. Better stick to the interface definitions :)

Reply:
Filed as #3828.